fe/fe-core/src/main/java/com/starrocks/planner/PlanNode.java

```diff
@@ -823,4 +833,14 @@ public boolean pushDownRuntimeFilters(RuntimeFilterDescription description, Expr
         }
         return false;
     }
+
+    public boolean canDoReplicatedJoin() {
+        if (children.size() == 1) {
+            return getChild(0).canDoReplicatedJoin();
+        } else if (children.size() == 2){
+            return getChild(1).canDoReplicatedJoin();
+        } else {
```

Comment:
Union_node is okay, but there is an implicit condition is union_node's children must be exchange

Reply:
Fixed